### PR TITLE
Fix update credit

### PR DESCRIPTION
**Pull request by jerry-yu**
fix judgement of update credit interval from eras to timstamp

### CODE REVIEW

File: pallets/credit/src/lib.rs [hunk 1]
```diff
@@ -497,6 +505,35 @@ pub mod pallet {
...
+            let diffs;
+            let mut era_used = false;
+            if let Some(pre_update_timestamp) = Self::last_credit_update_timestamp(server_id) {
+                let era_block_count = TryInto::<u64>::try_into(T::BlocksPerEra::get())
+                    .ok()
+                    .unwrap();
+                let secs_per_block = T::SecsPerBlock::get() as u64;
+                diffs = now_as_secs.saturating_sub(pre_update_timestamp)
+                    / era_block_count.saturating_mul(secs_per_block);
+            } else if let Some(last_credit_update_era) = Self::last_credit_update(&server_id) {
```
Comment (xcaptain) [inline pallets/credit/src/lib.rs:528]: Will this condition live forever or we can remove it in one or two months.

File: pallets/credit/src/lib.rs [hunk 2]
```diff
@@ -696,23 +738,22 @@ pub mod pallet {
...
-                if eras < 2 && Self::last_credit_update(&server_id).is_none() {
+                let now_as_secs = T::UnixTime::now().as_secs();
+                let (mut time_eras, era_used) = Self::check_update_credit_interval(
+                    &server_id,
+                    current_era,
+                    onboard_era.unwrap(),
+                    now_as_secs,
+                );
+
+                if time_eras < 2 && Self::last_credit_update_timestamp(&server_id).is_none() {
```
Comment (xcaptain) [inline pallets/credit/src/lib.rs:750]: looks duplicated, can we put this check into `check_update_credit_interval`?

**Review by xcaptain — COMMENTED** [review]
Overall looks good, would be better if you add more tests in `update_credit` and `add_credit_by_traffic` in tests.rs

File: pallets/credit/src/lib.rs (hunk 2 above)
Comment (jerry-yu) [reply pallets/credit/src/lib.rs:750]: it is for different usage. for new comming, giving them relexed update contition

**Review by xcaptain — APPROVED** [review]
LGTM

**Review by AmadeusGB — DISMISSED** [review]

**Comment by jerry-yu** [comment]
maybe use sessionid for era caculation is better

**Pull request closed without merging**